### PR TITLE
Resolve sorting and two Settings: default sort & short dots

### PR DESCRIPTION
**Pull request by ananzh**
### Description

* sorting

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/fb2b6230-093e-41d5-a258-e3d8d1f07a65

* default sort & short dots

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/36c5bd18-2612-4b92-8a12-7ef62091c4ce

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/discover/public/application/components/data_grid/data_grid_table.tsx [hunk 1]
```diff
@@ -149,32 +163,36 @@ export const DataGridTable = ({
   const legacyDiscoverTable = useMemo(
     () => (
       <LegacyDiscoverTable
-        displayedTableColumns={displayedTableColumns}
         columns={adjustedColumns}
         rows={rows}
         indexPattern={indexPattern}
-        sortOrder={sortingColumns}
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/components/data_grid/data_grid_table.tsx:156]: do we still need `sortingColumns`?

File: src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx [hunk 2]
```diff
@@ -87,15 +91,13 @@ export const LegacyDiscoverTable = ({
...
+              displayedColumns={displayedDefaultColumns}
+              defaultSortOrder={defaultSortOrder}
               indexPattern={indexPattern}
-              // isShortDots,
-              onChangeSortOrder={onChangeSortOrder}
+              onChangeSortOrder={onSort}
               onReorderColumn={onReorderColumn}
               onRemoveColumn={onRemoveColumn}
-              sortOrder={sortOrder}
+              sortOrder={sort as SortOrder[]}
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx:100]: is this needed? sort is already typed correctly above right?

File: src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx [hunk 3]
```diff
@@ -6,51 +6,55 @@
...
+  hideTimeColumn,
+  defaultSortOrder,
 }: DefaultDiscoverTableProps) => {
+  const displayedDefaultColumns = getLegacyDisplayedColumns(
+    columns,
+    indexPattern,
+    hideTimeColumn,
+    isShortDots
+  );
+  const adjustedDefaultColumnNames = displayedDefaultColumns.map((column) => column.name);
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx:57]:
```suggestion
  const displayedColumns = getLegacyDisplayedColumns(
    columns,
    indexPattern,
    hideTimeColumn,
    isShortDots
  );
  const displayedColumnNames = displayedColumns.map((column) => column.name);
```

File: src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx [hunk 4]
```diff
@@ -168,7 +144,7 @@ export function TableHeaderColumn({
         values: { columnName: name },
       }),
       className: 'fa fa-angle-double-left osdDocTableHeader__move',
-      onClick: () => onReorderColumn && onReorderColumn(name, currentIdx, colLeftIdx),
+      onClick: () => onReorderColumn && onReorderColumn(name, colLeftIdx),
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx:147]: If we have this, why are we making a new onMove function?

File: src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx [hunk 5]
```diff
@@ -13,24 +13,21 @@ import './_table_header.scss';
...
   colRightIdx: number; // idx of the column to the right, -1 if moving is not possible
   displayName: ReactNode;
   isRemoveable: boolean;
   isSortable?: boolean;
   name: string;
-  onChangeSortOrder?: (cols: EuiDataGridSorting['columns']) => void;
+  onChangeSortOrder?: (sortOrder: SortOrder[]) => void;
   onMoveColumn?: (name: string, idx: number) => void;
-  onReorderColumn?: (col: string, source: number, destination: number) => void;
+  onReorderColumn?: (colName: string, destination: number) => void;
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx:28]: Why do we have `onMove` and `onReorder`?

**Comment by ashwin-pc** [comment]
The only major comment i have is if we can reuse the onReorder function instead of creating a new one for move

File: src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx (hunk 5 above)
Comment (ananzh) [reply src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx:28]: remove this

File: src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx (hunk 4 above)
Comment (ananzh) [reply src/plugins/discover/public/application/components/default_discover_table/table_header_column.tsx:147]: you are right. will remove onMove.

File: src/plugins/discover/public/application/utils/state_management/common.ts
[the remark this answers is not in the document]
Comment (ananzh) [reply src/plugins/discover/public/application/utils/state_management/common.ts:1]: will add a test

File: src/plugins/discover/public/application/view_components/canvas/discover_table.tsx [hunk 6]
```diff
@@ -65,6 +66,10 @@ export const DiscoverTable = ({ rows }: Props) => {
     dispatch(reorderColumn({ source: source - 1, destination: destination - 1 }));
   };
 
+  const onMoveColumn = (colName: string, destination: number) => {
```
[the remark this answers is not in the document]
Comment (ananzh) [reply src/plugins/discover/public/application/view_components/canvas/discover_table.tsx:69]: I see this is the angular table using. For now, would like to make sort work by just restoring everything same. This moveColumn doesn't request a start idx.

File: src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx (hunk 2 above)
Comment (ananzh) [reply src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx:100]: will remove this type

**Pull request merged**